sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3942,6 +3943,69 @@ object functions {
    */
   def to_csv(e: Column): Column = to_csv(e, Map.empty[String, String].asJava)
 
+  // turn off style check that object names must start with a capital letter
+  // scalastyle:off
+  object partitioning {
```

Comment:
Does this has a Java corresponding test? This file should be compatible with Java side as well. I doubt if nested object works in Java API.

Reply:
This is a good point. We also have nested objects in `SQLConf` but it works fine. Maybe the java compatibility is the problem here.

Reply:
SQLConf is internal. I don't think it's supposed to be exposed.